src/query/functions/src/scalars/string_multi_args.rs

```diff
@@ -145,10 +141,7 @@ pub fn register(registry: &mut FunctionRegistry) {
                         }
                     }
 
-                    match len {
-                        Some(_) => Value::Column(Column::String(builder.build())),
-                        _ => Value::Scalar(Scalar::String(builder.build_scalar())),
-                    }
+                    Value::Column(Column::String(builder.build()))
```

Comment:
If the source argument is a `Scalar`, should the return value also be a `Scalar`?

Reply:
ok

Reply:
We can't determine whether the source argument is a `Scalar` or a `Column` based on `num_rows`, a `Column` may have only one row of values.

Reply:
If the column contains only one row, it's ok to return scalar.

We already support scalar and full columns in a `DataBlock`